Feature/no01

beforeisland opened the pull request:
NRIサイト、HTMLのコーディングが完了しました。

beforeisland committed:
featureブランチで2回目の編集をしました

beforeisland committed:
Initial Commit

beforeisland committed:
add:素材画像ファイルを保存

ページを横断して使われる画像をcommon/imegesに保存し、模写用ページ単体で使われる画像をimagesに保存しました。

beforeisland committed:
add:html<header>部分の実装

<header>部分を実装しました。グローバルナビ、サイドナビ、パンくずリストを実装。

beforeisland committed:
add:ボディ部分の実装

メイン記事から企業情報まで実装しました。

beforeisland committed:
add:ボディ中間部SNS部分の実装

ボディメイン部分とボディサブ部分に挟まれたSNS部分を実装しました。

beforeisland committed:
add:サブボディ部分(ローカルナビ)の実装

フッター上のローカルナビ部分を実装しました。

beforeisland committed:
add:フッター部分の実装

フッター部分のメニュー及び免責事項、トップページへ戻るボタンを実装しました。

beforeisland committed:
update:ヘッダー部fixメニュー部分修正

ヘッダー部の固定されているメニュー部分についてdivで囲みました。

beforeisland committed:
fix:リンク部分マークアップ間違いの修正

企業情報部分の<a>タグについて、文字列が終了タグの外に出ていたため、修正しました。

beforeisland committed:
fix:HTMLチェッカー使用後の修正

・右上3メニューへのalt属性追加
・記事部分のナビ項目について、見出しの文字列が入っていなかったため追記

beforeisland committed:
fix:HTMLチェッカー使用後の修正	2

・alt属性がaktと誤記されてしまっていたため修正

choimake commented on html-nri/index.html:
```diff
@@ -0,0 +1,341 @@
+                </nav>
+                <nav class="sideNavi">
+                    <div class="sideNavi-icon">
+                        <img src="common/images/glass.svg" width="20" height="20" alt="">
```
imgタグのaltには値を入れるようにしましょう。

beforeisland replied:
修正完了

update: imgタグのalt属性に値を付与	7a8fb6b	beforeisland <beforeisland@gmail.com>	2020/07/08 21:28

choimake commented on html-nri/index.html:
```diff
@@ -0,0 +1,341 @@
+                </nav>
+            </div>
+            <p>
+                <a href="https://www.nri.com/jp">NRI トップ</a> &gt; <a href="https://www.nri.com/jp/company">企業情報</a> &gt;
+                NRIの強み：NRIのデジタルビジネス戦略
```
カリキュラムでの記載が見当たらず申し訳ありませんが、今回は模写なのでhrefの値は「#」でお願いします。
（特に外部のページへ飛ぶ必要がないので）

beforeisland replied:
修正完了

fix: aタグhref属性のURL修正	bafbf6f	beforeisland <beforeisland@gmail.com>	2020/07/08 21:54

choimake commented on html-nri/index.html:
```diff
@@ -0,0 +1,341 @@
+                <li><a href="https://www.nri.com/jp/site/feedback">お問い合わせ</a></li>
+                <li><a href="https://www.nri.com/jp/company/map">アクセスマップ</a></li>
+            </ul>
+            <p id="copyright"><small>Copyright&copy; Nomura Research Institute, Ltd. All rights reserved.</small></p>
```
Copyrightの記号は「(c)」が正しいです。
（オリジナルのページはそうなってるので）

beforeisland replied:
修正完了

fix: copyright表記修正	894e786	beforeisland <beforeisland@gmail.com>	2020/07/08 21:57

choimake commented on html-nri/index.html:
```diff
@@ -0,0 +1,341 @@
+                </div>
+                <nav class="companyInfo">
+                    <h2>企業情報</h2>
+                    <div class="companyInfo-list">
```
ここは何にも意味を持たないdivよりも、区間を意味するsectionの方が使用するタグとしては適切だと考えます。

beforeisland replied:
(in my opinion)この部分はリンクの塊のため、意味的に整理されているものをマークアップするのであればsectionが良さそうですね。divはcssの装飾のために、複数の意味をもつ要素をひとまとめにしなければならない際に使いたいと思います。

beforeisland replied:
修正完了

fix: クラスcompanyInfo-listの要素をdivからsectionに変更	3b03556	beforeisland <beforeisland@gmail.com>	2020/07/09 6:47

choimake reviewed (commented):
コメントしましたので確認お願いします。

まずこちらから確認した限り、GitやGithubの使い方に関しては問題はありません。

commitのコメントも丁寧に記載されており、とても良いです。

改善点があるとすれば、
「fix:HTMLチェッカー使用後の修正」
「fix:HTMLチェッカー使用後の修正 2 」
というように同じような修正が並べるのは無駄にcommitが増えてしまって見にくいので、
修正内容に支障がなければ`git commit --amend`のようなコマンドを使って、
commitをひとつにまとめた方がより読みやすいです。

HTMLチェッカーでチェックをかけるとエラーがあったので、こちらは確実に修正してください。
カリキュラムにも記載されているように「HTMLチェッカーに通してみて、警告は出ない状態にしてから」提出するようにしましょう。

コードに関してのコメントは以下に続きます。

choimake commented on html-nri/index.html:
```diff
@@ -0,0 +1,341 @@
+            </p>
+
+        </header>
+        <div id="pageBody">
+            <div id="pageBodyMain">
+                <section class="mainVisual">
```
mainタグを使いましょう。

beforeisland replied:
修正完了

fix: id#pageBodyの要素をmainに変更	1b9a13e	beforeisland <beforeisland@gmail.com>	2020/07/09 6:52

choimake commented on html-nri/index.html:
```diff
@@ -0,0 +1,341 @@
+                    <p><a href="#" onclick="history.back(); return false;">前のページに戻る</a></p>
+                </nav>
+            </div>
+            <div id="pageBodysns">
```
idでの指定がされていますが、基本ただの装飾目的ならclassを使うのが一般的です。
idはJavaScriptでdomを操作する際に指定するパラメータで、単なる装飾目的では使いません。

beforeisland replied:
修正完了

fix: pageBodysnsをidからクラスへ変更	baebee8	beforeisland <beforeisland@gmail.com>	2020/07/09 6:56

choimake replied:
id -> class変換の適応範囲はページ全体の想定でした。
この一箇所だけではなくページ全体で、こちら対応お願いします。

beforeisland replied:
修正完了。ページ構成なども変更しましたので、コミットのコメントに詳細記載しました。

fix: id -> class変換及びタグ構成の整理	3e7165a	beforeisland <beforeisland@gmail.com>	2020/07/11 9:44

choimake reviewed (commented):
コメント追加しました。
合わせて確認お願いします。

beforeisland committed:
update: imgタグのalt属性に値を付与

ページ内の全てのimgタグのalt属性の値が空欄だったものを記述しました

beforeisland committed:
fix: aタグhref属性のURL修正

ページ内の全てのaタグhref属性のURLを#に修正しました

beforeisland committed:
fix: copyright表記修正

copyright表記を、元ページに倣い&and;から(c)に修正しました。

beforeisland committed:
fix: クラスcompanyInfo-listの要素をdivからsectionに変更

有機的に意味があるリストのため、クラスcompanyInfo-listの要素をdivからsectionに変更しました。

beforeisland committed:
fix: id#pageBodyの要素をmainに変更

id#pageBodyの要素をdivからmainに変更しました

beforeisland committed:
fix: pageBodysnsをidからクラスへ変更

単なる装飾目的の部分のため、pageBodysnsをidからクラスへ変更しました。

beforeisland committed:
fix: HTMLチェッカー使用後の修正

閉じタグ箇所ミス修正、articleタグ直下へのヘッダータグ追加

beforeisland commented:
修正完了のため、クローズ。

beforeisland commented:
コメント頂いた修正が完了しました。

beforeisland committed:
add: CSSファイル準備

CSSファイルを作成し、htmlへリンクしました。

choimake commented on html-nri/index.html:
```diff
@@ -0,0 +1,328 @@
+            <div id="pageBodySub">
+                <nav class="localNavi">
+                    <div class="localNavi-list">
+                        <h3><a href="#">ナレッジ・インサイト</a></h3>
+                        <ul>
+                            <li><a href="#">刊行物</a></li>
+                            <li><a href="#">書籍</a></li>
+                            <li><a href="#">コラム</a></li>
```
(in my opition)リンクにh3の見出しがついてますが、ここは文書構造的に見出しではないように思えます。
単純にリンクのままでよいのではないでしょうか？
（装飾はcssでやれば良いので）

以下のリスト全てに同様の適応箇所があるので見直してみてください。

beforeisland replied:
メインコンテンツ内の企業情報のリストと比較すると、指摘された箇所は明らかに見出しでは無いと認識しました。

そのため、指摘箇所の太字リンクはh3->pに修正しました。当初はpタグを使わずにaタグのままでもいいと思いましたが、最後段2個の太字リンクのみのリストについてはpタグ使用が望ましいと考えたため、他のリストも併せてpタグを入れています。


修正完了

fix: サブコンテンツの太字リンクをh3->pに変更	878551d	beforeisland <beforeisland@gmail.com>	2020/07/11 10:00

choimake commented on html-nri/index.html:
```diff
@@ -0,0 +1,328 @@
+                </nav>
+
+            </div>
+            <div class="pageBodysns">
```
(in my opinion)こことかもsectionあたりで良さそうですね。
divは装飾用途以外では使わず、可能な限り文章構造を示すタグを使うようにしましょう。

ページ全体で他にもdiv以外のタグで表現できそうなところがないか確認してみてください。
（僕視点だとたくさんあります）

beforeisland replied:
全体的に装飾用以外のものはセクショニングコンテンツに変更しました。



修正完了

fix: div->セクショニングコンテンツへの変更	7fe0734	beforeisland <beforeisland@gmail.com>	2020/07/11 10:30

beforeisland replied:
メインコンテンツとサブコンテンツの間のSNSリンク部分をsectionタグを使用して区分けしていましたが、見出しがないためセクショニングコンテンツで表現することが不適当と判断しました。また、そもそも装飾的にもsectionで囲む必要がなさそうなため、タグそのものを削除しました。


修正後

fix: htmlチェッカー使用後の修正	7568d60	beforeisland <beforeisland@gmail.com>	2020/07/11 10:46

choimake commented on html-nri/index.html:
```diff
@@ -0,0 +1,328 @@
+</head>
+
+<body>
+    <div id="page">
```
装飾目的のパラメータはidではなく、classを指定するようにしましょう。
ページ全体を通して、他にも同様の箇所がある場合は、修正してください。

beforeisland replied:
他の修正に含んで対応しました。タグ構成も一部変更しています。

fix: id -> class変換及びタグ構成の整理 3e7165a beforeisland beforeisland@gmail.com 2020/07/11 9:44

choimake commented on html-nri/index.html:
```diff
@@ -0,0 +1,328 @@
+                    <p class="articleLink-left">&lt; <a href="#">ビジネスモデルの特徴</a></p>
+                    <p class="articleLink-right"><a href="#">人材が支えるNRI</a> &gt;</p>
+
+                </article>>
```
タグの記述の仕方がおかしいです。

beforeisland replied:
修正完了

fix: article閉じタグミス修正	cb131e3	beforeisland <beforeisland@gmail.com>	2020/07/11 10:35

choimake reviewed (commented):
修正ありがとうございます。かなり良くなりました。

ただちょっと前回見落としていた部分でちょっと気になるところ、もう少し修正して欲しいところについて、コメントしています。
確認お願いします。

beforeisland committed:
add: ヘッダー部分のfloat実装

ヘッダー部分のサイトロゴ、グローバルナビ、サイドナビのfloatを実装しました。

beforeisland committed:
fix: id -> class変換及びタグ構成の整理

idは、ページの構成を[ヘッダー]、[メインコンテンツ]、[サブコンテンツ(フッター上のメニュー群)]、[フッター]の4つに分け、その部分のみに付与しました。[下部のページの先頭に戻る]については、headerに戻るように変更しています。
また、ページ全体を囲う#page、メインコンテンツとサブコンテンツを囲う#pageBodyは装飾用にもならず不要なタグと判断し削除しました。

beforeisland committed:
fix: サブコンテンツの太字リンクをh3->pに変更

サブコンテンツの太字リンクをh3->pに変更しました。当初aタグのままにしようかと思いましたが、最下部2個のリストはpタグ使用必須と考えたため、併せて他の6個のリスト上の太字リンクについてもpタグを付与しました。

beforeisland committed:
fix: div->セクショニングコンテンツへの変更

divタグは装飾用で使う部分のみに残し、セクショニングコンテンツに変更できる部分は変更しました。

beforeisland committed:
fix: article閉じタグミス修正

閉じタグの記述にミスがあったため修正しました。

beforeisland committed:
fix: htmlチェッカー使用後の修正

メインコンテンツとサブコンテンツの間のSNSリンク部分をsectionタグを使用して区分けしていましたが、見出しがないためセクショニングコンテンツで表現することが不適当と判断しました。また、そもそも装飾的にもsectionで囲む必要がなさそうなため、タグそのものを削除しました。

beforeisland commented:
修正完了しました。

choimake commented on html-nri/index.html:
```diff
@@ -0,0 +1,326 @@
+        </nav>
+
+    </main>
+    <div class="snsLink-typeB">
+        <p>Follow Us</p>
+        <ul>
+            <li>
+                <a href="#">
+                    <img src="common/images/icon-facebook.svg" width="24" height="24" alt="Facebookページ">
+                </a>
+            </li>
+            <li>
+                <a href="#">
+                    <img src="common/images/icon-twitter.svg" width="24" height="24" alt="Twitterページ">
+                </a>
+            </li>
+            <li>
+                <a href="#">
+                    <img src="common/images/icon-linkedin.svg" width="24" height="24" alt="linkedinページ">
+                </a>
+            </li>
+        </ul>
+    </div>
```
これはmainタグの中で良いと思います。
コンテンツは基本header、main、aside、footerのどれかに含まれているべきです。

beforeisland replied:
大枠のセクショニングに含まれている方が構造的に整理されている形になりますね。

修正完了

fix: snsリンクをmain内に移動	f26cf0f	beforeisland <beforeisland@gmail.com>	2020/07/12 19:08

choimake reviewed (commented):
修正ありがとうございます。
あと1件だけ対応して欲しいところがありましたので、その点についてコメントしてます。

あと文章構造について色々試行錯誤がみられるので、そのあたりを検証できるサイトをいくつか紹介します。まだ見たことがないようであれば確認してみてください。

HTML5 Outliner...htmlの文章構造がどうなっているか解析してくれるツールです。
https://gsnedders.html5.org/outliner/

whatwg...HTMLの仕様について色々策定しているグループです。タグの使い方などの情報が色々載っています。
https://html.spec.whatwg.org/multipage/

beforeisland committed:
fix: snsリンクをmain内に移動

.snsLink-typeBをmainタグ内に移動しました。
